tasks.py

```diff
@@ -86,35 +65,97 @@ def remove_ip_from_list(ip_address, ip_zone_name):
+    """
     container = apputils.Containerized()
-    target_date = apputils.Utility.iso8601_yesterday()
-    try:
-        container.scans_to_s3(target_date, s3_bucket_name)
-    except Exception as e:
-        "Exception encountered: %s" % e
-        raise self.retry(countdown=120, exc=e, max_retries=5)
+    return container.generic_container_launch_attached(image,
+                                                       env_literal.copy(),
+                                                       env_expand.copy(),
+                                                       read_only)
 
 
 @app.task(bind=True)
-def prior_day_events_to_s3(self, s3_bucket_name):
+def generic_bound_containerized_task(self, image, env_literal, env_expand,
+                                     retry, log_messages, read_only=False):
+    """Wrap Containerized.generic_container_launch_attached() for scheduler.
+
+    This task is a generic interface for scheduled tasks to launch containers.
+
+    Args:
+        image(str): Container image to launch.
+        env_literal(dict): Dictionary containing environment variables to be
+            passed into the container launch configuration, unmodified.
+        env_expand(dict): Dictionary containing arguments to be expanded
+            from environment variables, then added to the environment variables
+            defined in env_literal.  The value for every key in the dictionary
+            is replaced by the corresponding environment variable.  For
+            instance, with this dictionary: ``{'API': 'API_KEY'}``, and
+            the existence of an environment variable ``API_KEY`` set to
+            ``abc123``, before launching the container, the dictionary will be
+            processed and the result will be ``{'API': 'abc123'}``.  The result
+            will be added to the environment variables used to launch the
+            container.
+        retry(int): Number of times to retry after failure.
+        log_messages(dict): This contains the log messages used to indicate
+            the activities and success or failure of the task.  Expected keys
+            in this dictionary include ``task_started``, ``task_finished``,
+            ``task_retried``, and ``task_failed``.
+    """
+    start_msg = log_messages["task_started"]
+    finished_msg = log_messages["task_finished"]
+    retried_msg = log_messages["task_retried"]
+    fail_msg = log_messages["task_failed"]
     container = apputils.Containerized()
-    target_date = apputils.Utility.iso8601_yesterday()
     try:
-        container.events_to_s3(target_date, s3_bucket_name)
+        print("TaskRunner: %s" % start_msg)
+        container.generic_container_launch_attached(image, env_literal.copy(),
+                                                    env_expand.copy(),
+                                                    read_only)
+        print("TaskRunner: %s" % finished_msg)
     except Exception as e:
```

Comment:
could we please specify the error we are catching here? it's not efficient to retry on all errors